pkg/node/types/node.go

```diff
@@ -65,12 +66,13 @@ func ParseCiliumNode(n *ciliumv2.CiliumNode) (node Node) {
 		Cluster:         option.Config.ClusterName,
 		ClusterID:       option.Config.ClusterID,
 		Source:          source.CustomResource,
-		Labels:          n.ObjectMeta.Labels,
 		Annotations:     n.ObjectMeta.Annotations,
 		NodeIdentity:    uint32(n.Spec.NodeIdentity),
 		WireguardPubKey: wireguardPubKey,
 	}
 
+	node.Labels = labelsfilter.FilterLabelsByRegex(option.Config.ExcludeNodeLabelPatterns, n.ObjectMeta.Labels)
```

Comment:
So, I also wanted to update data stored in kvstore under `cilium/state/nodes/v1/*`
IIRC from few months ago, I remember adding in more places to make sure exclude patterns are applied to kvstore as well. I can re-test this on a real cluster and get back to you on this.

Reply:
Thanks! The logic to update the CiliumNode resource and the equivalent kvstore representation got [recently refactored and uniformed](https://github.com/cilium/cilium/pull/29319), and they now both directly retrieve the source of truth from the [LocalNodeStore updates](https://github.com/cilium/cilium/blob/0554ca68117599dae110ff1e9c7d01731d534f2e/pkg/nodediscovery/nodediscovery.go#L233-L235), which sources them from the [corresponding k8s Node resource](https://github.com/cilium/cilium/blob/0554ca68117599dae110ff1e9c7d01731d534f2e/daemon/cmd/local_node_sync.go#L226-L230).

Reply:
Given Marco's comment, should the filtration of the labels be in the "ingestion" layer here as soon as the object is received or should it be somewhere that's more clearly the "processing" layer? I think it would be useful for debugging if Cilium knew what the original event was and what the processed event is.

@giorio94, WDYT? Do you have thoughts on where this code should be?

Reply:
I'm personally in favor of keeping the filtering logic as part of the `ParseNode` function, both for consistency with the already existing filtering logic for annotations and to make sure that all consumers observe the same object version. Otherwise, we risk inconsistencies between let's say the propagation of the information into the CiliumNode and the corresponding kvstore representation, which already occurred in the past and caused subtle bouncing issues. I'd not be particularly concerned from the debugging perspective, given that this would be an opt in flag meant mainly for fine tuning, and it would perform a pretty straightforward filtering operation.

Reply:
@giorio94 updated the PR to invoke the filtering logic only on `ParseNode`. Removed the flag and doc updates from the operator as well.